### PR TITLE
UMessage: validator and unit test

### PR DESCRIPTION
**Pull request by billpittman**
UMessage: Validator and unit test.

### CODE REVIEW

**Comment by billpittman** [comment]
@gregmedd Sorry I didn't see your message until this morning ... closed my browser.

File: test/coverage/datamodel/UMessageValidatorTest.cpp [hunk 1]
```diff
@@ -11,28 +11,764 @@
...
+	        .count();
+
+	id.set_msb((timestamp << 16) | (8ULL << 12) |
+	           (0x123ULL));  // version 8 ; counter = 0x123
+	id.set_lsb((2ULL << 62) | (0xFFFFFFFFFFFFULL));  // variant 10
+
+	attributes.set_type(UMESSAGE_TYPE_REQUEST);
+	attributes.set_allocated_id(new UUID(id));
+	attributes.set_allocated_source(new UUri(source));
+	attributes.set_allocated_sink(new UUri(sink));
```
Comment (gregmedd) [inline test/coverage/datamodel/UMessageValidatorTest.cpp:73]: Don't use `set_allocated_` or `new`

File: test/coverage/datamodel/UMessageValidatorTest.cpp [hunk 2]
```diff
@@ -11,28 +11,764 @@
...
+		source_.set_resource_id(1);
+
+		sink_.set_authority_name("10.0.0.2");
+		sink_.set_ue_id(0x00010002);
+		sink_.set_ue_version_major(2);
+		sink_.set_resource_id(2);
+
+		reqId_.set_msb(0x1234);
+		reqId_.set_lsb(0x5678);
+	}
```
Comment (gregmedd) [inline test/coverage/datamodel/UMessageValidatorTest.cpp:38]: If these don't change, do they need to be initialized for each individual test?

File: test/coverage/datamodel/UMessageValidatorTest.cpp [hunk 3]
```diff
@@ -11,28 +11,764 @@
...
+	UAttributes attributes;
+
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+
+	id.set_msb((timestamp << 16) | (8ULL << 12) |
+	           (0x123ULL));  // version 8 ; counter = 0x123
+	id.set_lsb((2ULL << 62) | (0xFFFFFFFFFFFFULL));  // variant 10
```
Comment (gregmedd) [inline test/coverage/datamodel/UMessageValidatorTest.cpp:92]: Could you just use the UUID builder here?

**Review by gregmedd — COMMENTED** [review]
There are definitely some minor issues that could use clean-up in this PR. However, I don't know that any of the comments I'm adding are severe enough to block this change right now. I am considering capturing the feedback as issues and merging this code to unblock downstream work (e.g. up-transport-zenoh-cpp).

**Comment by gregmedd** [comment]
None of the changes I requested would prevent us from moving forward. I'll file an immediate follow-up PR to address these changes since Bill is not currently available.

File: src/datamodel/validator/UMessage.cpp [hunk 4]
```diff
@@ -10,3 +10,345 @@
...
+		}
+	}
+
+	if (!UPriority_IsValid(umessage.attributes().priority())) {
+		return {false, Reason::PRIORITY_OUT_OF_RANGE};
+	}
+
+	if (!UPayloadFormat_IsValid(umessage.attributes().payload_format())) {
+		return {false, Reason::PAYLOAD_FORMAT_OUT_OF_RANGE};
+	}
```
[the remark this answers is not in the document]
Comment (pranavishere2) [reply src/datamodel/validator/UMessage.cpp:101]: Thanks for suggestion. I have now used this in #161

**Pull request merged**